rdbms/src/main/java/org/spine3/server/storage/jdbc/table/AbstractTable.java

```diff
@@ -0,0 +1,337 @@
+        return query;
+    }
+
+    /**
+     * Flags that the {@linkplain #getIdColumnDeclaration() ID column} should be declared as
+     * a {@code PRIMARY KEY}.
+     *
+     * <p>Override to change the table creation behavior.
+     *
+     * @return {@code true} by default
+     */
+    protected boolean idIsPrimaryKey() {
+        return true;
+    }
+
+    protected Logger log() {
+        return logger;
+    }
+
+    private String composeCreateTableSql() {
+        final String idColumnName = getIdColumnDeclaration().name();
+        @SuppressWarnings("StringBufferReplaceableByString")
+        final StringBuilder sql = new StringBuilder(MEAN_SQL_QUERY_LENGTH);
+        sql.append(CREATE_IF_MISSING)
+           .append(getName())
+           .append(BRACKET_OPEN);
+        for (C column : getColumns()) {
+            final String name = column.name();
+            final Sql.Type type = ensureType(column);
+            sql.append(name)
+               .append(type)
+               .append(COMMA);
+            // Comma after the last column declaration is required since we add PRIMARY KEY after
+        }
+        if (idIsPrimaryKey()) {
+            sql.append(PRIMARY_KEY)
+               .append(BRACKET_OPEN)
+               .append(idColumnName)
+               .append(BRACKET_CLOSE);
+        }
+        sql.append(BRACKET_CLOSE)
+           .append(SEMICOLON);
+        final String result = sql.toString();
+        return result;
+    }
+
+    private Sql.Type getIdType() {
+        final Sql.Type idType = getIdColumn().getColumnDataType();
+        return idType;
+    }
+
+    /**
+     * For the ID column checks the {@link Sql.Type} and returns the type of the ID column if
+     * the ID type is {@linkplain Sql.Type#UNKNOWN}.
+     *
+     * <p>If the column does not represent an ID of the table, this method throws
+     * an {@link IllegalStateException}.
+     *
+     * @param column the column the type of which will be checked
+     * @return the SQL type of the column
+     */
+    private Sql.Type ensureType(C column) throws IllegalStateException {
+        Sql.Type type = column.type();
+        if (type == Sql.Type.UNKNOWN) {
+            if (column == getIdColumnDeclaration()) {
+                type = getIdType();
+            } else {
+                throw new IllegalStateException("UNKNOWN type of a non-ID column " + column.name());
+            }
+        }
+        return type;
+    }
+
+    /**
+     * Deletes a row in the table corresponding to the given ID.
+     *
+     * @param id ID to search by
+     * @return {@code true} if the row was deleted successfully, {@code false} if the row was
+     * not found
+     */
+    public boolean delete(I id) {
+        final DeleteRecordQuery<I> query =
+                DeleteRecordQuery.<I>newBuilder()
+                                 .setTableName(getName())
+                                 .setIdColumn(getIdColumn())
+                                 .setIdColumnName(
```

Comment:
It looks like `IdColumn` can have a name to remove the need to have two methods instead of one.